src/clients/python/crequest.cc

```diff
@@ -103,27 +117,32 @@ struct ServerHealthContextCtx {
 nic::Error*
 ServerHealthContextNew(
     ServerHealthContextCtx** ctx, const char* url, int protocol_int,
-    bool verbose)
+    const char** headers, int num_headers, bool verbose)
 {
   nic::Error err;
-  ProtocolType protocol;
-  err = ParseProtocol(&protocol, protocol_int);
-  if (err.IsOk()) {
-    ServerHealthContextCtx* lctx = new ServerHealthContextCtx;
-    if (protocol == ProtocolType::HTTP) {
-      err = nic::ServerHealthHttpContext::Create(
-          &(lctx->ctx), std::string(url), verbose);
-    } else {
-      err = nic::ServerHealthGrpcContext::Create(
-          &(lctx->ctx), std::string(url), verbose);
-    }
 
+  std::map<std::string, std::string> http_headers;
+  err = ParseHttpHeaders(&http_headers, headers, num_headers);
```

Comment:
Shouldn't parse headers be done after checking that HTTP protocol is used?

Reply:
Yes, fixed